### PR TITLE
4_data_pipeline

### PR DESCRIPTION
**Pull request by Jason-Perez27**
1_band_names.py: Get a text file of band names for each TIF image.
2_geojson_bbox.py: Converts TIF bounding boxes into GeoJSON polygons in EPSG:4326.
3_dataset_C02.py: Merges TIF data and annotations by extracting band values for each labeled point.
4_column_names.py: Renames raw band columns to meaningful names (e.g., Red, Blue) based on Landsat version.
5_decode_qa_bands.py: Decodes Collection 2 QA bitmasks and appends QA flags as new columns.
6_qa_cloud_filter.py: Filters out rows flagged with clouds or invalid data using QA columns.
7_data_cleaning.py: Removes unneeded columns and optionally retains ST_* columns for a final cleaned dataset.
8_label_id.py: Adds a numeric “Label ID” column grouping ‘Irr’ vs ‘Unirr’ classes.
9_train_test_split.py: Splits the cleaned dataset into a 70/30 train/test set, grouping by TIF ID.

### CODE REVIEW

File: code/4_data_pipeline/3_dataset_C02.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+                        # Append data to list
+                        tif_id = tif_name_to_id[key]
+                        row = [
+                            tif_id, tif_filename, landsat_version, year, month, day,
+                            x_percent, y_percent, label, x, y
+                        ] + band_data
+                        data.append(row)
+
+# Step 5: Write data to Excel file
```
Comment (anna-boser) [inline code/4_data_pipeline/3_dataset_C02.py:142]: saving as excel is a bit unconventional -- why not just a csv?

Comment (anna-boser) [reply code/4_data_pipeline/3_dataset_C02.py:142]: It's not a big deal I mean this works it's just not usual. Next time I would say use a csv.

File: code/4_data_pipeline/5_decode_qa_bands.py
Comment (anna-boser) [inline code/4_data_pipeline/5_decode_qa_bands.py:1]: Why are there two decode_qa_band files?

File: code/4_data_pipeline/8_label_id.py
Comment (anna-boser) [inline code/4_data_pipeline/8_label_id.py:1]: This is small but I find the name of this file confusing. It's not really a label id -- it's more just a binary label. I'd call this 8_binary_label or something

**Review by anna-boser — CHANGES_REQUESTED** [review]
Again, add the descriptions of the data to data.txt and a quick one liner (like you have in the description of the pull request) to the headers of the files. As you add stuff to data.txt, also add it to the waves folder on the cluster though filezilla or vs code or whatever is easiest so we make sure everything is synched. 

Otherwise this generally looks good and clean except I'm not sure why there are two files for decoding the qa bands.
